cmd/kubelet/app/server.go

```diff
@@ -770,10 +770,16 @@ func run(ctx context.Context, s *options.KubeletServer, kubeDeps *kubelet.Depend
 		if err != nil {
 			return fmt.Errorf("--kube-reserved value failed to parse: %w", err)
 		}
+		if kubeReserved != nil && !isExactMilli(kubeReserved[v1.ResourceCPU]) {
+			klog.InfoS("kubeReserved cpu is not a integer(in millicores), which may cause frequent node updates", "cpu", s.KubeReserved[v1.ResourceCPU.String()])
```

Comment:
IMO, This log should be a `Warning` instead of `Info`

In addition to printing this log, would it be better to round the `resource.Quantity` up to avoid frequent update issue?

Additional, will putting those logic in to `parseResourceList` be better?

Reply:
> This log should be a `Warning` instead of `Info`

the migration-to-structured-logging doc says:
```
klog.Warning, klog.Warningf, klog.Warningln -> klog.InfoS
```
ref: https://github.com/kubernetes/community/blob/master/contributors/devel/sig-instrumentation/migration-to-structured-logging.md#change-log-functions

> In addition to printing this log, would it be better to round the `resource.Quantity` up to avoid frequent update issue?

In addition to reporting node allocatable resources, the reserve config may also be used for setting underlying cgroup resource? Not sure where to do the round and minimize the impact of the modification.

> Additional, will putting those logic in to `parseResourceList` be better?

 `parseResourceList` may be referenced elsewhere in the future? Both are ok for me.

Reply:
1. My bad, please ignore me.
2. This looks like doing round in apiserver, see [comment here](https://github.com/kubernetes/kubernetes/pull/126835#issuecomment-2301581636), the very specific place need additional investigation. AFAIK, may not be completely accurate, kubelet will not modify the root cgroup(`kubepods.slice`). additional, allocatable resources are only awared by scheduler, but the pod cgroups are related with limit and request in `pod.spec`. 
3. Since this function is unexported, IMO it's fine.